src/Riok.Mapperly/Descriptors/Mappings/TypeMappingBuildContext.cs

```diff
@@ -63,6 +63,14 @@ SyntaxFactoryHelper syntaxFactory
         return (ctx, scopedSourceName);
     }
 
+    public (TypeMappingBuildContext Context, string SourceName) WithNewScopedSource(string sourceName)
+    {
+        var scopedNameBuilder = NameBuilder.NewScope();
+        var scopedSourceName = scopedNameBuilder.New(sourceName);
+        var ctx = new TypeMappingBuildContext(IdentifierName(scopedSourceName), ReferenceHandler, scopedNameBuilder, SyntaxFactory);
+        return (ctx, scopedSourceName);
+    }
```

Comment:
Instead of this new method, an optional parameter could be added to the existing `WithNewScopedSource()` method: `WithNewScopedSource(string sourceName = DefaultSourceName)` and `WithNewScopedSource(Func<string, ExpressionSyntax> sourceBuilder, string sourceName = DefaultSourceName)`